demisto_sdk/commands/common/hook_validations/id.py

```diff
@@ -290,8 +293,11 @@ def is_entity_version_match_playbook_version(self, implemented_entity_list_from_
             bool. Whether the playbook's version match playbook's entities.
         """
         invalid_version_entities = []
-        is_valid = True
+        implemented_entities = implemented_entity_list_from_playbook.copy()
```

Comment:
use `implemented_entities` across the loop (lines 304 and 406) - as we remove items, the search in the list will be faster each time.